cluster-autoscaler/core/utils/utils.go

```diff
@@ -36,25 +36,30 @@ import (
 )
 
 // GetNodeInfoFromTemplate returns NodeInfo object built base on TemplateNodeInfo returned by NodeGroup.TemplateNodeInfo().
-func GetNodeInfoFromTemplate(nodeGroup cloudprovider.NodeGroup, daemonsets []*appsv1.DaemonSet, ignoredTaints taints.TaintKeySet) (*schedulerframework.NodeInfo, errors.AutoscalerError) {
+func GetNodeInfoFromTemplate(nodeGroup cloudprovider.NodeGroup, daemonsets []*appsv1.DaemonSet, ignoredTaints, unwantedTaints taints.TaintKeySet) (*schedulerframework.NodeInfo, errors.AutoscalerError) {
 	id := nodeGroup.Id()
 	baseNodeInfo, err := nodeGroup.TemplateNodeInfo()
 	if err != nil {
 		return nil, errors.ToAutoscalerError(errors.CloudProviderError, err)
 	}
 
-	labels.UpdateDeprecatedLabels(baseNodeInfo.Node().ObjectMeta.Labels)
+	sanitizedBaseNodeInfo, typedErr := SanitizeNodeInfo(baseNodeInfo, id, ignoredTaints, unwantedTaints)
```

Comment:
Should we really SanitizeNodeInfo twice? Why would we need to go through deep copying the node twice, generate 2 random names for it, etc? The performance impact is negligible, but it seems confusing and potentially error prone with extra temporary deep copies.

I think here what we really need is just SanitizeNode()? And than we can decouple it from sanitizing pods that we would need to do at the end of this function?

Reply:
Nice catch, changed to SanitizeNode and setting it on the baseNodeInfo.